FirebaseVertexAI/Sample/FunctionCallingSample/ViewModels/FunctionCallingViewModel.swift

```diff
@@ -103,7 +106,7 @@ class FunctionCallingViewModel: ObservableObject {
   func startNewChat() {
     stop()
     error = nil
-    chat = model.startChat()
+    chat = nil
```

Comment:
Weird that `chat` is set to nil in `startNewChat`

Reply:
This allowed me to leave calling `await model.startChat()` in `sendMessage`, which was already `async`. I've now refactored `ConversationViewModel` and `FunctionCallingViewModel` to work around this.